Subsonicious/Subsonicious/Views/LoginView.swift

```diff
@@ -34,39 +35,64 @@ struct LoginView: View {
                         .textContentType(.password)
                         .asLoginField()
                 }
+                .disabled(isLoading)
 
                 Button(action: continueButtonPressed) {
-                    Text("login.continue")
-                        .font(.headline)
-                        .foregroundColor(.white)
-                        .frame(maxWidth: content.size.width, maxHeight: 44)
-                        .background(Color.yellow)
-                        .cornerRadius(6)
-                        .opacity(continueButtonOpacity)
+                    if isLoading {
+                        ProgressView()
```

Comment:
why the if inside the button and not instead of ?

```
if isLoading {
    ProgressView()
} else {
    Button(action: continueButtonPressed) {
          Text("login.continue")
                             .font(.headline)
                             .foregroundColor(.white)
                             .frame(maxWidth: .infinity, maxHeight: .infinity)
    }
}

```

Reply:
Because I want the ProgressView to be part of the Button and to replace the Text :) The same way some other apps do ;)